src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -392,6 +393,218 @@ internal string WorkingDirectory
+        /// </param>
+        /// <returns>
+        /// Returns true if the argument was parsed successfully and false if not.
+        /// </returns>
+        private static bool ParseSettingFileHelper(string[] args, int argIndex, CommandLineParameterParser parser)
+        {
+            if (argIndex >= args.Length)
+            {
+                if (parser != null)
+                {
+                    parser.WriteCommandLineError(
+                        CommandLineParameterParserStrings.MissingSettingsFileArgument);
+                }
+
+                return false;
+            }
+
+            string configFile = null;
+            try
+            {
+                configFile = NormalizeFilePath(args[argIndex]);
+            }
+            catch (Exception ex)
+            {
+                if (parser != null)
+                {
+                    string error = string.Format(CultureInfo.CurrentCulture, CommandLineParameterParserStrings.InvalidSettingsFileArgument, args[argIndex], ex.Message);
+                    parser.WriteCommandLineError(error);
+                }
+
+                return false;
+            }
+
+            if (!System.IO.File.Exists(configFile))
+            {
+                if (parser != null)
+                {
+                    string error = string.Format(CultureInfo.CurrentCulture, CommandLineParameterParserStrings.SettingsFileNotExists, configFile);
+                    parser.WriteCommandLineError(error);
+                }
+
+                return false;
+            }
+            PowerShellConfig.Instance.SetSystemConfigFilePath(configFile);
+            return true;
+        }
+
+        /// <summary>
+        /// Processes the command line parameters to ConsoleHost which must be parsed before the Host is created.
+        /// Success to indicate that the program should continue running.
+        /// </summary>
+        /// <param name="args">
+        /// The command line parameters to be processed.
+        /// </param>
+        internal static void EarlyParse(string[] args)
+        {
+            Dbg.Assert(!_dirtyEarlyParse, "This instance has already been used. Create a new instance.");
+
+            // indicates that we've called this method on this instance, and that when it's done, the state variables
+            // will reflect the parse.
+            _dirtyEarlyParse = true;
+
+            EarlyParseHelper(args);
+        }
+
+        private static string GetConfigurationNameFromGroupPolicy()
+        {
+            // Current user policy takes precedence.
+            var consoleSessionSetting = Utils.GetPolicySetting<ConsoleSessionConfiguration>(Utils.CurrentUserThenSystemWideConfig);
+            if (consoleSessionSetting != null)
+            {
+                if (consoleSessionSetting.EnableConsoleSessionConfiguration == true)
+                {
+                    if (!string.IsNullOrEmpty(consoleSessionSetting.ConsoleSessionConfigurationName))
```

Comment:
We could combine three `if` in one or use `?` operator in just single `return`.

Reply:
`?` only checks for null and could have regressions.